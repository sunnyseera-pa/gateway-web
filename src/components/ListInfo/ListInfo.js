/** @jsx jsx */
import { cx } from '@emotion/css';
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import LayoutBox from '../LayoutBox';
import { PROP_TYPES_LAYOUTBOX } from '../LayoutBox/LayoutBox.propTypes';
import * as styles from './ListInfo.styles';

const ListInfo = ({ data, className, widthCol1, mt, mb, ml, mr, width, minWidth, maxWidth }) => {
	return (
		<LayoutBox {...{ mt, mb, ml, mr, width, minWidth, maxWidth }}>
			<ul css={styles.root} className={cx('ui-ListInfo', className)}>
				{data.map(({ label, value }, i) => (
					<li css={styles.listItem} key={`${i}-${label}`}>
						<div css={styles.col1(widthCol1)}>{label}</div>
						<div>{value}</div>
					</li>
				))}
			</ul>
		</LayoutBox>
	);
};

ListInfo.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			label: PropTypes.node,
			value: PropTypes.node,
		})
	).isRequired,
	widthCol1: PropTypes.string,
	...PROP_TYPES_LAYOUTBOX,
};

ListInfo.defaultProps = {
	widthCol1: '105px',
};

export default ListInfo;
