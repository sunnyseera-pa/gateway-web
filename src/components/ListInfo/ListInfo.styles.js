import { css } from '@emotion/react';

export const root = css`
	list-style-type: none;
	padding: 0;
	margin: 0;
`;

export const listItem = css`
	display: flex;
	margin-bottom: 12px;

	&:last-child {
		margin: 0;
	}
`;

export const col1 = widthCol1 => theme => css`
	color: ${theme.colors.grey700Alt};
	min-width: ${widthCol1};
	max-width: ${widthCol1};
	margin-right: 32px;
`;
