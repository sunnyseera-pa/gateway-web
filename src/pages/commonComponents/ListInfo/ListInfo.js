/** @jsx jsx */
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import * as styles from './ListInfo.styles';

const ListInfo = ({ data, widthCol1 }) => {
	return (
		<ul css={styles.root}>
			{data.map(({ label, value }) => (
				<li css={styles.listItem}>
					<div css={styles.col1(widthCol1)}>{label}</div>
					<div>{value}</div>
				</li>
			))}
		</ul>
	);
};

ListInfo.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			label: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
			value: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
		})
	).isRequired,
	widthCol1: PropTypes.string,
};

ListInfo.defaultProps = {
	widthCol1: '105px',
};

export default ListInfo;
