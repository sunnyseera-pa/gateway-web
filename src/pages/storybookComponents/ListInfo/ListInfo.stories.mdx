import { Meta, Story, Canvas } from '@storybook/addon-docs';
import iconTick from '../../../images/tick.svg';
import iconRejected from '../../../images/Application_rejected.svg';

import ListInfo from './ListInfo';

<Meta
	title='Core/ListInfo'
	component={ListInfo}
	args={{
		data: [
			{
				label: 'Previous answer',
				value: 'The NHFD FA is an annual survey of all trauma centres in the UK.',
			},
			{
				label: 'Updated answer',
				value:
					'The NHFD FA is an annual survey of all trauma centres in the UK. It includes data on the support for clinical governance, organisation of theatres and care bundles that are used. Full datasets available on request: NHFD@rcplondon.ac.uk ',
			},
		],
	}}
/>

<Canvas>
	<Story name='Basic'>{props => <ListInfo {...props} />}</Story>
</Canvas>
