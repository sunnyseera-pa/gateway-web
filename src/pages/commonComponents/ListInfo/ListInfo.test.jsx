import { render } from '@testing-library/react';
import React from 'react';
import ListInfo from '.';

let wrapper;

const props = {
	className: 'additional-classname',
	data: [
		{
			label: 'Label 1',
			value: 'Value 1',
		},
		{
			label: 'Label 2',
			value: 'Value 2',
		},
	],
};

describe('Given the ListInfo component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<ListInfo {...props} />, { wrapper: Providers });
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then has the correct className', () => {
			expect(wrapper.container.querySelector('.additional-classname')).toBeTruthy();
		});

		it('Then contains the correct label', () => {
			expect(wrapper.getByText('Label 1')).toBeTruthy();
			expect(wrapper.getByText('Label 2')).toBeTruthy();
		});

		it('Then contains the correct label', () => {
			expect(wrapper.getByText('Value 1')).toBeTruthy();
			expect(wrapper.getByText('Value 2')).toBeTruthy();
		});
	});
});
