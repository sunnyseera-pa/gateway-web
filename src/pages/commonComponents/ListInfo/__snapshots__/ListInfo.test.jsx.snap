// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ListInfo component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <ul
    class="additional-classname css-1jupbku-ListInfo"
  >
    <li
      class="css-oc8kcp"
    >
      <div
        class="css-112bk1i"
      >
        Label 1
      </div>
      <div>
        Value 1
      </div>
    </li>
    <li
      class="css-oc8kcp"
    >
      <div
        class="css-112bk1i"
      >
        Label 2
      </div>
      <div>
        Value 2
      </div>
    </li>
  </ul>
</div>
`;
